### DISCUSSION TITLE
Create connection in UI to conect to azure sql database

### DISCUSSION THREAD

**Discussion by d0dulk0** [apache/airflow#31041] [Q&A]
Apache Airflow version
2.5.3

Hello team ! 
I am using the official prod image apache/airflow:2.5.3-python3.10 as the base for my Dockerfile
I have checked the docu for provider apache-airflow-providers-microsoft-azure
[ https://airflow.apache.org/docs/apache-airflow-providers-microsoft-azure/stable/index.html](url)

There doesn't seem to be an out of the box connector to Azure SQL Database here, so I am wondering what do I have to do to create such connection in the airflow UI

I have tried to do it via ODBC by filling 
Host, Schema, Login, Password, Port, but I get an error :

`('IM002', '[IM002] [unixODBC][Driver Manager]Data source name not found and no default driver specified (0) (SQLDriverConnect)')
`

**Comment by potiuk** [comment]
No idea to be honest - but maybe you should try to reach out to Azure support of their managed airflow? They recently released manafed service and likely have some guidelines there. Generally the other Cloud providers (Google and AWS) are actively contributing maintaining, adding features here and responding to questions but we have not seen Azure or Microsoft doing the same, so if there is anything missing for Azure integration, I would rather redirect you to their support - because you are not likely to receive any help here from them.

**Reply by d0dulk0** [reply]
Hello Jarek!
thank you so much for the information. So do I understand it correctly that the provider packages are developed/maintained by the actual providers and not by the airflow community ? 

I have done some digging and have found that ODBC should indeed be used to connect to MSSQL (which I think Azure SQL DB is essentially that) according to the documentation https://airflow.apache.org/docs/apache-airflow-providers-odbc/stable/connections/odbc.html

So I will try to set it up and post my progress here.

**Reply by potiuk** [reply]
> thank you so much for the information. So do I understand it correctly that the provider packages are developed/maintained by the actual providers and not by the airflow community ?

No - they are maintined by the community, to the extend we have them, but we have also mixed governance model: https://github.com/apache/airflow/blob/main/PROVIDERS.rst#mixed-governance-model (and by the way this is the doc that you can learn all about how providers are maintained) which means that we closely collaborate with the service providers who chose to collaborate with us. That includes for example THIS dashboard from AWS https://aws-mwaa.github.io/open-source/system-tests/dashboard.html which amazon MWAA team maintains and for example this PR where people backed by  Google Composer team lead for the last few months to make sure all the google clients are up-to-date 
  https://github.com/apache/airflow/pull/30067

And this means that if there are missing or unclear things when it comes to those providers, we have people we can @-mention here from those teams and they are REALLY REALLY helpful at times.

I wish (I REALLY wish) I could say the same about Azure team. But sadly, I can only say that they are largetly MIA even if Azure released Managed Airflow quite a few months ago. So the best I can can do in this case is to direct you to their support team.

`¯\_(ツ)_/¯`
